### ISSUE TITLE
getMediaWithTagName ERROR 400

### ISSUE THREAD

**Issue by Natelegreat1** [shyambhat/InstagramKit#150]
Hi! I'm trying to use getMediaWithTagName but keep getting the following error. Not sure wha tot make of it?
    
    Error Domain=com.alamofire.error.serialization.response Code=-1011 "Request failed: bad request (400)" UserInfo=0x7ff6f28a72b0 {com.alamofire.serialization.response.error.response=<NSHTTPURLResponse: 0x7ff6f2877ac0> { URL: https://api.instagram.com/v1/tags/hi/media/recent?client_id=2708926af8fd46d288e579286855ab1e&count=0 } { status code: 400, headers {
    "Cache-Control" = "private, no-cache, no-store, must-revalidate";
    Connection = "keep-alive";
    "Content-Language" = en;
    "Content-Length" = 112;
    "Content-Type" = "application/json; charset=utf-8";
    Date = "Mon, 07 Sep 2015 18:43:05 GMT";
    Expires = "Sat, 01 Jan 2000 00:00:00 GMT";
    Pragma = "no-cache";
    "Set-Cookie" = "csrftoken=abc35833eefff1dd4a0a0fc927faf164; expires=Mon, 05-Sep-2016 18:43:05 GMT; Max-Age=31449600; Path=/";
    Vary = "Cookie, Accept-Language";
    "X-Ratelimit-Limit" = 5000;
    "X-Ratelimit-Remaining" = 4996;
    } }, NSErrorFailingURLKey=https://api.instagram.com/v1/tags/hi/media/recent?client_id=2708926af8fd46d288e579286855ab1e&count=0, com.alamofire.serialization.response.error.data=<7b226d65 7461223a 7b226572 726f725f 74797065 223a2241 5049496e 76616c69 64506172 616d6574 65727345 72726f72 222c2263 6f646522 3a343030 2c226572 726f725f 6d657373 61676522 3a22436f 756e7420 6d757374 20626520 6c617267 65722074 68616e20 7a65726f 2e227d7d>, NSLocalizedDescription=Request failed: bad request (400)}

**Comment by manumax** [comment]
I tried to manually perform your call and the response was the following:
```swift
{"meta":{"error_type":"APIInvalidParametersError","code":400,"error_message":"Count must be larger than zero."}}
```
Hopefully you've already solved, but a very easy solution would be to use `getMediaWithTagName:count:maxId:withSuccess: failure:` instead.

That said, in my opinion it would be nice to enhance a bit how `InstagramKit` handle this kind of errors. @shyambhat Wouldn't be better to wrap API error inside a more specific ` InstagramKit`'s `NSError`? To clarify, something like:
```Objective-C
NSError *error = [NSError errorWithDomain:InstagramKitErrorDomain
                                     code:[whatever]
                                 userInfo:@{
                                     NSLocalizedDescriptionKey: @"Count must be larger than zero.",
                                     RawError:  [error returned by AFNetworking]
                                 }];
```

**Comment by ezefranca** [comment]
+1

**Comment by shyambhat** [comment]
Fixed in #169
